### PR TITLE
Cleanup dependency use and separate example dependencies

### PR DESCRIPTION
**Pull request by je-cook**
## Description

As discussed earlier today, separate out examples dependencies from the main dependencies

## Interface Changes

None, example dependencies are still installed on environment setup, just separated out into distinct files

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: requirements-conda.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+cftime==1.6.0
+gmsh-interop==2021.1.1
+h5py==3.3.0
+importlib-metadata==4.11.3
+pybind11==2.9.2
+pybind11-global==2.9.2
+PySide2==5.13.2
+pytools==2022.1.7
+PyYAML==6.0
+Shapely==1.8.2
```
Comment (CoronelBuendia) [inline requirements-conda.txt:12]: Think now that BLUEPRINT is gone, shapely may no longer be directly required. Not sure if it is needed by any other dependencies though.

Comment (je-cook) [reply requirements-conda.txt:12]: Yeah I tried that and something pulls it in (possibly meshio..)

**Review by hsaunders1904 — APPROVED** [review]

**Comment by je-cook** [comment]
To keep a clean history for both develop and develop_dependencies this will go into develop_dependencies and come in with release 2

File: conda/environment.yml [hunk 2]
```diff
@@ -22,16 +22,17 @@ dependencies:
   - python=3.8
   - freecad=0.19.3
   - pivy=0.6.5 # later pivy versions (0.6.7 at the time of writing) crash our FreeCAD CAD viewer
-  - numpy=1.21.5
+  - numpy=1.21.6
```
Comment (CoronelBuendia) [inline conda/environment.yml:25]: Should we be changing versions of deliberate dependencies in this PR?

File: setup.py [hunk 3]
```diff
@@ -55,25 +53,28 @@
...
 
 prominence = [
     "prominence",
 ]
 
+pinned = [
+    "nlopt==2.7.0",
+    "numba==0.55.1",
+    "numba-scipy==0.3.0",
+    "numpy==1.21.6",
```
Comment (CoronelBuendia) [inline setup.py:67]: See above

**Review by CoronelBuendia — APPROVED** [review]
LGTM although I don't consider myself qualified to review properly! Just a minor question

File: conda/environment.yml (hunk 2 above)
Comment (je-cook) [reply conda/environment.yml:25]: Oops that must have come in via something automatic, either way this should be changed to reflect 1.22.4 that went in the other day. Will fix, thanks

**Pull request merged**